### PR TITLE
Add Timelens support, to improve video navigation capabilities

### PR DESCRIPTION
**Pull request by blinry**
This adds support for displaying visual timelines and popup thumbnails
throughout voctoweb's interface, aiming to make video navigation more
efficient, precise, and active. See <https://timelens.io> for an explanation of
what this is all about.

Two fields, "timeline_filename" and "thumbnails_filename" are added to
the Event model. When present, the Timelens plugin for MediaElement.js
is activated. Also, Timelens interfaces are added to video listings.

When the Timelens fields are not filled, everything works as before.

### CODE REVIEW

File: app/controllers/api/events_controller.rb [hunk 1]
```diff
@@ -83,7 +83,7 @@ def set_event
   def create_event(params)
     @event.transaction do
       @event.release_date = Time.now unless @event.release_date
-      @event.set_image_filenames(params[:thumb_url], params[:poster_url])
+      @event.set_image_filenames(params[:thumb_url], params[:poster_url], params[:timeline_url], params[:thumbnails_url])
```
Comment (derchrisuk) [inline app/controllers/api/events_controller.rb]: Should be 

`@event.get_image_filenames(params[:thumb_url], params[:poster_url], params[:timeline_url], params[:thumbnails_url])`

Comment (blinry) [reply app/controllers/api/events_controller.rb]: Oh! "set" is correct here, but by mistake, my commit changed the name of the method in the Event model to "get".  Fixed and updated this branch!

File: app/models/frontend/event.rb [hunk 2]
```diff
@@ -146,6 +154,10 @@ def relive
       conference.metadata['relive']&.first { |r| r['guid'] == guid }
     end
 
+    def timelens_present?
+        not timeline_filename.blank? and not thumbnails_filename.blank?
```
Comment (manno) [inline app/models/frontend/event.rb]: Da `present?` das [Gegenteil von !blank?](https://apidock.com/rails/Object/present%3F) ist, würde ich die positive Form bevorzugen: 
`timeline_filename.present? and thumbnails_filename.present?`

Comment (blinry) [reply app/models/frontend/event.rb]: Good point! Fixed and force-updated the branch. Feel free to merge this PR!

**Comment by manno** [comment]
@blinry I can't deploy anymore, maybe this is related to this PR?

```
ubuntu@media:/vagrant$ RACK_ENV=production rails assets:precompile
I, [2018-10-03T12:54:46.088184 #3048]  INFO -- : Writing /vagrant/public/assets/embed-d8402f8bd4ff35a95fde03727c761eca5c46e3e08e12647e312501feed6cf174.css
I, [2018-10-03T12:54:46.090756 #3048]  INFO -- : Writing /vagrant/public/assets/embed-d8402f8bd4ff35a95fde03727c761eca5c46e3e08e12647e312501feed6cf174.css.gz
rails aborted!
Uglifier::Error: Unexpected token: name (vtt_url). To use ES6 syntax, harmony mode must be enabled with Uglifier.new(:harmony => true).
/var/lib/gems/2.3.0/gems/uglifier-4.1.14/lib/uglifier.rb:234:in `parse_result'
/var/lib/gems/2.3.0/gems/uglifier-4.1.14/lib/uglifier.rb:216:in `run_uglifyjs'
/var/lib/gems/2.3.0/gems/uglifier-4.1.14/lib/uglifier.rb:168:in `compile'
```

**Pull request merged**